daemon/media_socket.c

```diff
@@ -45,6 +45,9 @@
 #define MAX_RECV_LOOP_STRIKES 5
 #endif
 
+#define AUTO_DIRECTION_PREFIX "auto-to-"
+#define AUTO_DIRECTION_PREFIX_LEN (sizeof(AUTO_DIRECTION_PREFIX) - 1)
```

Comment:
Have you considered using a dedicated dictionary key for this, instead of re-using the interface selection key with a magic prefix string? It's fine either way, but perhaps it would make for a cleaner interface.

So e.g. for interface selection we have `direction=` or alternatively `from-interface=` and `to-interface=`, or just `interface=` for other methods. And then to use address-based interface selection, you could have `from-address=` and `to-address=` or something similar. Just an idea, haven't looked at how much more trouble this would be.